app/components/computed-props.js

```diff
@@ -50,11 +51,88 @@ export default Component.extend({
    * Please replace these props below with computed props
    * @type {Array}
    */
-  userNames: [],
-  bmwusers: [],
-  whiteCarsUsers: [],
-  usersWith1Car: [],
-  userswithMoreThanOneCar: [],
-  allMakes: [] // no repeat of makes
+  userNames: computed('users', function () {
+    const userNames = [];
+    this.get('users').forEach((item) => {
+      userNames.push(item.name);
+    });
+    return userNames;
+  }),
+  bmwusers: computed('users', function () {
+    const bmwUsers = [];
+    this.get('users').forEach((user) => {
+      const foundCar = user.cars.find((car) => car.make === 'BMW');
+      if (foundCar) {
+        bmwUsers.push(user.name);
+      }
+    });
+    return bmwUsers;
+  }),
+  whiteCarsUsers: computed('users', function () {
+    const whiteCarUsers = [];
+    this.get('users').forEach((user) => {
+      const foundCar = user.cars.find((car) => car.color === 'white');
+      if (foundCar) {
+        whiteCarUsers.push(user.name);
+      }
+    });
+    return whiteCarUsers;
+  }),
+  usersWith1Car: computed('users', function () {
+    const usersWith1Car = [];
+    this.get('users').forEach((user) => {
+      if (user.cars.length === 1) {
+        usersWith1Car.push(user.name);
+      }
+    });
+    return usersWith1Car;
+  }),
+  userswithMoreThanOneCar: computed('users', function () {
+    const usersWithMoreThan1Car = [];
+    this.get('users').forEach((user) => {
+      if (user.cars.length > 1) {
+        usersWithMoreThan1Car.push(user.name);
+      }
+    });
+    return usersWithMoreThan1Car;
+  }),
+  allMakes: computed('users', function () {
+    const makeOfCar = [];
+    this.get('users').forEach((user) => {
+      user.cars.forEach((car) => {
+        if (!makeOfCar.includes(car.make)) {
+          makeOfCar.push(car.make);
+        }
+      });
+    });
+    return makeOfCar;
+  }),
+  newCarColor: '',
+  actions: {
+    setNewCarColor(color) {
+      this.set('newCarColor', color);
+    },
+    addNewUser() {
+      this.get('users').pushObject({
```

Comment:
that would be where your `data` is mutated think.

To avoid this, you would need to clone `data`. You have `Ember.copy`, or other forms of cloning using pure JS available.

Reply:
I think instead of `this.set('users', A(data));`

you would need to clone `data`, `Ember.copy` or something

Reply:
Yes I had to manually copy it as Ember.copy wouldn't let me do a deep copy of data

Reply:
was thinking on how to simplify this code. Could use `computed.merge` === `computed.union` to merge the two arrays (initial user array + added users). But fundamentally, the problem you face is that you need an ember-data `user` model and create new instances of it, which is not in the scope of this bootcamp app (strangely) so let's not spend too much time on this in here.